pkg/util/restoptions/helpers.go

```diff
@@ -51,10 +52,14 @@ func ApplyOptions(optsGetter Getter, store *registry.Store, oldIsNamespaced bool
 	if store.CreateStrategy == nil {
 		return fmt.Errorf("store for %s must have CreateStrategy set", store.QualifiedResource.String())
 	}
-
-	isNamespaced := store.CreateStrategy.NamespaceScoped()
-	if isNamespaced != oldIsNamespaced { // TODO(soltysh): oldIsNamespaced should be completely removed in #12541
-		return fmt.Errorf("CreateStrategy has %v for namespace scope but user specified %v as namespace scope", isNamespaced, oldIsNamespaced)
+	if store.ObjectNameFunc == nil {
```

Comment:
So there is no PR upstream yet.  Basically I am saying that we plan to replace `ApplyOptions` with `store.CompleteWithOptions`.  Thus `store.CompleteWithOptions` must at least do everything `ApplyOptions` does.  Therefore it needs to be updated to default a missing `ObjectNameFunc`.  At that point it would make sense to delete the `ObjectNameFunc`s in upstream as they would no longer be useful.  Since all of this is will be a relatively large change to the storage layer, it would make sense for us to upstream the etcd storage path test to give us higher confidence that we did not accidentally break something.